lib/matplotlib/colorbar.py

```diff
@@ -699,10 +705,12 @@ def _do_extends(self, ax=None):
             # add the patch
             val = 0 if self._long_axis().get_inverted() else -1
             color = self.cmap(self.norm(self._values[val]))
+            hatch_idx = len(self._y) - 1
             patch = mpatches.PathPatch(
                 mpath.Path(xy), facecolor=color, alpha=self.alpha,
                 linewidth=0, antialiased=False,
-                transform=self.ax.transAxes, hatch=hatches[-1], clip_on=False,
+                transform=self.ax.transAxes, hatch=hatches[hatch_idx],
+                clip_on=False,
```

Comment:
I'm missing it; is this not the same as before?

Reply:
No, hatches is "arbitrary" long (we extend it opportunistically to make sure it is long enough) so the index should be determined based on the length of `self._y`.  If we did something like `hatches = make_same_length(hatches, self._y)` one could use the last index, but that is not the case now.

Reply:
Ah good point yes, that's fine.